### PR TITLE
fix: patch jclouds 2.5.0 with important fix for google-cloud-storage; forbid double quotes in resource name

### PR DESCRIPTION
**Pull request by Maxim-Gadalov**

### CODE REVIEW

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/487301)

File: build.gradle [hunk 1]
```diff
@@ -45,8 +45,13 @@ dependencies {
     implementation 'com.fasterxml.jackson.core:jackson-databind:2.15.2'
     implementation 'com.auth0:java-jwt:4.4.0'
     implementation 'com.auth0:jwks-rsa:0.22.1'
-    implementation 'org.apache.jclouds:jclouds-allblobstore:2.5.0'
-    implementation 'org.apache.jclouds.api:filesystem:2.5.0'
+    // fat jar of https://github.com/Maxim-Gadalov/jclouds/tree/2.5.0-gcp-fix
```
Comment (astsiapanay) [inline build.gradle]: I suggested to replace your forked repo with an official repo and branch where the fat jar was built on.

File: src/main/java/com/epam/aidial/core/storage/ResourceDescription.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
 public class ResourceDescription {
 
-    private static final Set<Character> INVALID_FILE_NAME_CHARS = Set.of('/', '{', '}');
+    private static final Set<Character> INVALID_FILE_NAME_CHARS = Set.of('/', '{', '}', '"');
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/storage/ResourceDescription.java:22]: Is that a part of the patch?
should we add this fix to the develop branch as well ?

Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/storage/ResourceDescription.java:22]: this fix was requested for this specific version

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/487316)

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/487355)

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/487361)

File: build.gradle (hunk 1 above)
Comment (Maxim-Gadalov) [reply build.gradle]: fixed

**Comment by Maxim-Gadalov** [comment]
/deploy-review

>Environment URL: https://chat-ai-dial-core-pr-375.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-375.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by astsiapanay — APPROVED** [review]

**Pull request merged**